File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        hash_map = {}
        
        for num in nums:
            if num in hash_map:
                hash_map[num] += 1
            else:
                hash_map[num] = 1
        
        for key in hash_map:
            if hash_map[key] % 2:
                return False
        
        return True

File: 2206-divide-array-into-equal-pairs/2206-divide-array-into-equal-pairs.go
func divideArray(nums []int) bool {
    hashMap := make(map[int]int)
    
    for i := range nums {
        _, ok := hashMap[nums[i]]
        if ok {
            hashMap[nums[i]]++
        } else {
            hashMap[nums[i]] = 1
        }
    }
    
    for _, v := range hashMap {
        if v%2 == 1 {
            return false
        }
    }
    return true
}